system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,370 @@
+            $file = new SplFileInfo($file);
+        }
+
+        $fileContent = file_get_contents($file->getRealPath());
+        preg_match_all('/lang\(\'([._a-z0-9\-]+)\'\)/ui', $fileContent, $matches);
+
+        if (empty($matches[1])) {
+            return [];
+        }
+
+        foreach ($matches[1] as $phraseKey) {
+            $phraseKeys = explode('.', $phraseKey);
+
+            /**
+             * Language key not have Filename or Lang key
+             */
+            if (count($phraseKeys) < 2) {
+                continue;
+            }
+
+            $languageFileName   = array_shift($phraseKeys);
+            $isEmptyNestedArray = (! empty($languageFileName) && empty($phraseKeys[0]))
+                || (empty($languageFileName) && ! empty($phraseKeys[0]))
+                || (empty($languageFileName) && empty($phraseKeys[0]));
+
+            if ($isEmptyNestedArray) {
+                continue;
+            }
+
+            /**
+             * Language key as string
+             */
+            if (count($phraseKeys) === 1) {
+                /**
+                 * Add found language keys to temporary array
+                 */
+                $languageFoundKeys[$languageFileName][$phraseKeys[0]] = $phraseKey;
+            } else {
+                $childKeys                            = $this->buildMultiArray($phraseKeys, $phraseKey);
+                $languageFoundKeys[$languageFileName] = array_replace_recursive($languageFoundKeys[$languageFileName] ?? [], $childKeys);
+            }
+        }
+
+        return $languageFoundKeys;
+    }
+
+    private function isIgnoredFile(SplFileInfo $file): bool
+    {
+        if ($file->isDir() || str_contains($file->getRealPath(), $this->languagePath)) {
+            return true;
+        }
+
+        return 'php' !== $file->getExtension();
+    }
+
+    private function templateFile(array $language = []): string
+    {
+        if (! empty($language)) {
+            $languageArrayString = $this->languageKeysDump($language);
+
+            return <<<PHP
+                <?php
+
+                return {$languageArrayString};
+
+                PHP;
+        }
+
+        return <<<'PHP'
+            <?php
+
+            return [];
+
+            PHP;
+    }
+
+    /**
+     * Create multidimensional array from another keys
+     */
+    private function buildMultiArray(array $fromKeys, string $lastArrayValue = ''): array
+    {
+        $newArray  = [];
+        $lastIndex = array_pop($fromKeys);
+        $current   = &$newArray;
+
+        foreach ($fromKeys as $value) {
+            $current[$value] = [];
+            $current         = &$current[$value];
+        }
+
+        $current[$lastIndex] = $lastArrayValue;
+
+        return $newArray;
+    }
+
+    /**
+     * Compare recursive two arrays and return new array (diference)
+     */
+    private function arrayDiffRecursive(array $originalArray, array $compareArray): array
```

Comment:
This method is not tested at all.
It seems better to extract the code to a class, and add more test cases.

Reply:
There is no suitable place in the system except Helpers. Move the code to a function? 
I thought about it before writing, but decided to leave it in the class. Without creating an extra function for one class

Reply:
For example, create `CodeIgniter\Commands\Translation\ArrayDiff`.
If we will need it elsewhere, we may consider making it a helper.

Reply:
Done!

Reply:
Thank you!